src/pages/guides/getting-started/creating-your-first-plugin/index.md

```diff
@@ -0,0 +1,182 @@
+
+The _ps-starter_ template initially contains a panel with a _"Populate Layers"_ button that allows you to populate a box
+within the panel with the layer names:
+
+![A Photoshop panel with a list of layer names and a button called "Populate Layers"](panel-base.png)
+
+Let's take a look at what's going on in the background when running this plugin.
+
+In the `manifest.json` file, the `"main"` field points to the `index.html` file:
+
+```json
+{
+  "id": "my.first.plugin",
+  "name": "My first plugin",
+  "version": "1.0.0",
+  "main": "index.html",
+  "host": [
+    /* ... */
+  ],
+  "manifestVersion": 4,
+  "entrypoints": [
+    {
+      "type": "panel",
+      "id": "vanilla",
+      // [...]
+      "icons": [
+        // [...]
+      ],
+      "label": {
+        "default": "Starter Panel"
+      }
+    }
+  ],
+  "icons": [
+    // [...]
+  ]
+}
+```
+
+When Photoshop loads the plugin, it loads the `index.html` file specified here. The `index.html` file, in turn, defines
+our panel's UI:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <script src="index.js"></script>
+  </head>
+  <style>
+    /* [...] */
+  </style>
+  <body>
+    <sp-heading>Layers</sp-heading>
+    <sp-body id="layers"> No layers </sp-body>
+    <footer>
+      <sp-button id="btnPopulate">Populate Layers</sp-button>
+    </footer>
+  </body>
+</html>
+```
+
+As you can see, the `index.html` file then loads the `index.js`.
+
+<InlineAlert variant="info" slots="text"></InlineAlert>
+
+Note that while this looks like fairly standard HTML, the UXP environment is not a browser and only supports a [subset of HTML/CSS/JS features](/src/pages/uxp/reference-html/General/Unsupported%20Elements.md). You can also see some custom components like `<sp-heading>`, `<sp-body>`, and `<sp-button>`. These are so-called [Spectrum UXP components](/src/pages/uxp/reference-spectrum/index.md) that implement the Spectrum Design system and allow plugins to feel native in the surrounding Photoshop environment.
+
+The `index.js` file then sets up a click listener on the _"Populate
+Layers"_ button and uses the Photoshop APIs to populate the `<sp-body id="layers" />` element with the layer names:
+
+```js
+function showLayerNames() {
+  const app = window.require("photoshop").app;
+  const allLayers = app.activeDocument.layers;
+  const allLayerNames = allLayers.map((layer) => layer.name);
+  const sortedNames = allLayerNames.sort((a, b) =>
+    a < b ? -1 : a > b ? 1 : 0
+  );
+  document.getElementById("layers").innerHTML = `
+      <ul>${sortedNames.map((name) => `<li>${name}</li>`).join("")}</ul>`;
+}
+
+document
+  .getElementById("btnPopulate")
+  .addEventListener("click", showLayerNames);
+```
+
+Note that you can use the `window.require()` function to import various API modules, like `require('photoshop')` in this
+example.
+
+## Step 3: Adjust the plugin
+
+_If we were lazy, we could already call it a day. But let's face it: This doesn't feel like you've built your own
```

Comment:
Let's rephrase slightly -- maybe `We could call this good enough, but let's face it....` :-)